File: code/java/removeNthFromEnd.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode p = new ListNode(0);
        ListNode dummy = new ListNode(0);
        ListNode p2 = new ListNode(0);
        
        p.next = head;
        p2.next = head;
        dummy.next = p;
        int total = 0;
        int i = 0;
        
        while(p2.next != null){
            total ++;
            p2 = p2.next;
        }
        
        while(p != null){
            if(i+n==total)
                p.next = p.next.next;
            i++;
            p = p.next;
        }
        
        return dummy.next.next;
    }
}

File: code/python/removeNthFromEnd.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def removeNthFromEnd(self, head: ListNode, n: int) -> ListNode:
        p = ListNode(0)
        dummy = p
        p.next = head
        p2 = ListNode(0)
        p2.next = head
        total = 0
        i = 0

        while p2.next:
            total += 1
            p2 = p2.next
        
        while p:
            if i == total - n:
                p.next = p.next.next
            i+=1
            p = p.next
                
        return dummy.next
